backend/worker/autoresolver.go

```diff
@@ -0,0 +1,85 @@
+package worker
+
+import (
+	"context"
+	"time"
+
+	"github.com/highlight-run/highlight/backend/model"
+	privateModel "github.com/highlight-run/highlight/backend/private-graph/graph/model"
+	"github.com/highlight-run/highlight/backend/store"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+type AutoResolver struct {
+	store *store.Store
+	db    *gorm.DB
+}
+
+func NewAutoResolver(store *store.Store, db *gorm.DB) *AutoResolver {
+	return &AutoResolver{
+		store: store,
+		db:    db,
+	}
+}
+
+func (autoResolver *AutoResolver) AutoResolveStaleErrors(ctx context.Context) {
+	projectFilterSettings := autoResolver.store.FindProjectsWithAutoResolveSetting(ctx)
+
+	for _, projectFilterSettings := range projectFilterSettings {
+		log.WithContext(ctx).WithFields(
+			log.Fields{"project_id": projectFilterSettings.ProjectID}).Info("Finding stale errors for project")
+		interval := projectFilterSettings.AutoResolveStaleErrorsDayInterval
+
+		project, err := autoResolver.store.GetProject(ctx, projectFilterSettings.ProjectID)
+		if err != nil {
+			log.WithContext(ctx).WithFields(log.Fields{"project_id": projectFilterSettings.ProjectID}).Error(err)
+			continue
+		}
+
+		err = autoResolver.resolveStaleErrorsForProject(ctx, project, interval)
+
+		if err != nil {
+			log.WithContext(ctx).WithFields(log.Fields{"project_id": projectFilterSettings.ProjectID}).Error(err)
+			continue
+		}
+	}
+}
+
+func (autoResolver *AutoResolver) resolveStaleErrorsForProject(ctx context.Context, project model.Project, interval int) error {
+	var errorGroups []model.ErrorGroup
+
+	db := autoResolver.db
+
+	err := db.Debug().
+		Table("error_groups").
+		Select("DISTINCT(error_groups.id), error_groups.project_id").
+		Joins("LEFT JOIN error_objects ON error_groups.id = error_objects.error_group_id").
```

Comment:
this should be an `inner join`, since we search for `error_objects.created_at` which is `null` if there are no `error_objects` resulting from the `left join`

Reply:
Good catch. Also, looks like we were missing the project_id in the where clause. 
I fixed both in d6dadd54a

The generated query:

```sql
SELECT DISTINCT
	(error_groups.id),
	error_groups.project_id
FROM
	"error_groups"
	INNER JOIN error_objects ON error_groups.id = error_objects.error_group_id
WHERE (error_groups.state = 'OPEN')
	AND(error_groups.project_id = 1)
	AND(error_objects.created_at < '2023-04-25 09:48:15.667')
```

runs in 47.160s